dirs/dirs.go

```diff
@@ -199,10 +198,13 @@ func init() {
 	SetRootDir(root)
 }
 
-func GetSnapHomeDirs() []string {
+// An array is passed by reference in order to avoid mutating snapHomeDirs
+// The array content is cleared first to avoid undesirable elements in the slice
+func GetSnapHomeDirs(arr *[]string) {
```

Comment:
I think the previous signature was slightly better
```suggestion
func GetSnapHomeDirs() []string {
	snapHomeDirsMu.Lock()
	defer snapHomeDirsMu.Unlock()
	dirs := make([]string, len(snapHomeDirs))
	copy(dirs, snapHomeDirs)
}
```

Reply:
oh never mind my bad, for some reason I assumes that `dirs` is a global variable and not directly defined in `GetSnapHomeDirs`...